addons/purchase_stock/models/purchase.py

```diff
@@ -254,24 +255,25 @@ def _create_picking(self):
         return True
 
     def _update_date_planned_for_lines(self, updated_dates):
-        note = self._compose_note(updated_dates)
+        """Override to add picking info in the activity note.
+        """
+        activity = self._create_update_date_activity(updated_dates)
 
         validated_picking = self.picking_ids.filtered(lambda p: p.state == 'done')
+        changed_pattern = re.compile("<p>Those dates have.*?</p>")
+        unchanged_pattern = re.compile("<p>Those dates couldn’t.*?</p>")
         if validated_picking:
-            note += _("<p>Those dates couldn’t be modified accordingly on the receipt %s which had already been validated.</p>") % validated_picking.name
+            unchanged_line = _("<p>Those dates couldn’t be modified accordingly on the receipt %s which had already been validated.</p>") % validated_picking[0].name
```

Comment:
same here, getting back the pattern is painful so maybe replace totally the previous activity or create a new one with the error message?